cmd/kube-scheduler/app/server.go

```diff
@@ -168,11 +171,18 @@ func Run(ctx context.Context, cc schedulerserverconfig.CompletedConfig, registry
 		}
 	}
 
+	// Prepare event clients.
+	if _, err := cc.Client.Discovery().ServerResourcesForGroupVersion(eventsv1beta1.SchemeGroupVersion.String()); err == nil {
+		cc.Broadcaster = events.NewBroadcaster(&events.EventSinkImpl{Interface: cc.EventClient.Events("")})
+		cc.Recorder = cc.Broadcaster.NewRecorder(scheme.Scheme, cc.ComponentConfig.SchedulerName)
+	}
```

Comment:
I expected:
```go
if _, err := cc.Client.Discovery().ServerResourcesForGroupVersion(eventsv1beta1.SchemeGroupVersion.String()); err == nil {
  cc.Broadcaster = events.NewBroadcaster(&events.EventSinkImpl{Interface: cc.EventClient.Events("")})
} else {
  cc.Broadcaster = ... v1 events-based implementation of events.EventBroadcaster
}
cc.Recorder = cc.Broadcaster.NewRecorder(scheme.Scheme, cc.ComponentConfig.SchedulerName)
```

Reply:
with something like this:
```
type v1RecorderWrapper struct {
	recorder record.EventRecorder
}

func (w v1RecorderWrapper) Eventf(regarding runtime.Object, related runtime.Object, eventtype, reason, action, note string, args ...interface{}) {
	w.recorder.Eventf(regarding, eventtype, reason, note, args...)
}
```

Reply:
+1 to what Jordan wrote
Especially given that Recorder interface is so simple that it has only 1 method:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/events/interfaces.go#L25

Reply:
I opened https://github.com/kubernetes/kubernetes/pull/84442 - @yastij feel free to patch it into your PR